File: 122. Best Time to Buy and Sell Stock II/122. Best Time to Buy and Sell Stock II.js
/**
 * @param {number[]} prices
 * @return {number}
 */
function maxProfit(prices) {
    let totalProfit = 0;
    
    for (let i = 1; i < prices.length; i++) {
        if (prices[i] > prices[i - 1]) {
            totalProfit += prices[i] - prices[i - 1];
        }
    }
    
    return totalProfit;
}

console.log(maxProfit([7,1,5,3,6,4]));  // Output: 7
console.log(maxProfit([1,2,3,4,5]));    // Output: 4
console.log(maxProfit([7,6,4,3,1]));    // Output: 0

File: 122. Best Time to Buy and Sell Stock II/122. Best Time to Buy and Sell Stock II.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut total_profit = 0;
        
        for i in 1..prices.len() {
            let profit = prices[i] - prices[i-1];
            if profit > 0 {
                total_profit += profit;
            }
        }
        
        total_profit
    }
}